### ISSUE TITLE
Swapping keys / key-combinations

### ISSUE THREAD

**Issue by DennisSchiefer** [tekezo/Karabiner-Elements#559]
I've been unsuccessfully trying to accomplish two remappings --or rather swapping of two key combinations-- with Karabiner Elements alone or in combination with Better Touch Tool, and Keyboard Maestro:

1. Swap Command-Tab and Control-Tab
Background: I swapped Command and Control in macOS System Settings, but I would like to retain the same physical keys for switching between applications and tabs.
I realize that Karabiner Elements currently cannot handle this so I was looking into Better Touch Tool and Keyboard Maestro. From Keyboard Maestro guides I gathered that swapping was not possible as both bindings would interfere. So I tried Better Touch Tool, but even implementing just one direction did not work. Which brings me to the conclusion that the old Karabiner (where this remapping worked) changes keys on a lower system level than Better Touch Tool, and I hope that Karabiner Elements does the same.

2. Swap Command-H with Command-Option-Control-H
Backgound: I often accidently hit Command-H and hide my current window and I just want this shortcut disabled.
Again, Karabiner Elements cannot do the trick at the moment. I was successful in disabling Command-H with Better Touch Tool, but again swapping two key combinations was not possible.

In conclusion: Is there any way at the moment to swap key combinations with some tool?

**Comment by trevorsenior** [comment]
This may be related to https://github.com/tekezo/Karabiner-Elements/issues/558 / they were opened around the same time. Did you just update macOS? I'm running:

<img width="244" alt="image" src="https://cloud.githubusercontent.com/assets/1592403/22002433/16abf898-dc1a-11e6-8357-8f792829911c.png">

And Karabiner-Elements:

<img width="237" alt="image" src="https://cloud.githubusercontent.com/assets/1592403/22002455/30560284-dc1a-11e6-81bb-d5c3fbe820d6.png">

And all of my key swaps broke once macOS updated.

**Comment by trevorsenior** [comment]
Removing everything and re-installing from scratch fixed this.

- Using Misc --> Launch uninstaller (in Karabiner).
- Reinstall the latest version.
- Re-add the entry for Karabiner-Elements in Security & Privacy if it's missing
- If the "simple modifications" are still there after reinstall, remove them all and re-add them.

**Comment by patrickhulce** [comment]
I too am in desperate need of swapping key combinations, specifically for PC-style Home/End behavior. 
It looks like the single key code dependency goes all the way down to [event_manipulator.hpp](https://github.com/tekezo/Karabiner-Elements/blob/master/src/core/grabber/include/manipulator/event_manipulator.hpp)? Not sure if it's as simple as expanding to an arbitrary list<key_code> and threading it through everywhere

**Comment by DennisSchiefer** [comment]
FYI, I am now using https://ei-kana.appspot.com/ (as referenced elsewhere in the issue tracker here) for the two rebindings that I needed.

**Comment by uniquejava** [comment]
@DennisSchiefer Thank you for recommendation, finally I find a substitute of Karabiner,  though I can't read Japanese, but that app is quite simple to use and just what I need. I just want to simply change Cmd + H to Ctrl + H and Cmd + Q to something else.  That app is simple and straightforward.

And it's written in Swift3. Awesome!

**Comment by mmazzarolo** [comment]
@DennisSchiefer thank you for the recommendation, that software fixed my `alt+tab` and `ctrl+tab` swapping in a breeze :)

**Comment by DDvO** [comment]
Meanwhile it is possible to swap key combinations with Karabiner Elements complex modifications.
The trick seems to be to use two separate rules, one for each direction.
I leared this from the example at https://github.com/pqrs-org/Karabiner-Elements/issues/3084#issuecomment-1123586212

For instance, I swap command-tab and option-tab by adding two `rules` elements this way:
```
                    {
                        "description": "map command-tab to option-tab",
                        "manipulators": [
                            {
                                "from": {
                                    "key_code": "tab",
                                    "modifiers": {
                                        "mandatory": [
                                            "command"
                                        ],
                                        "optional": [
                                            "any"
                                        ]
                                    }
                                },
                                "to": [
                                    {
                                        "key_code": "tab",
                                        "modifiers": [
                                            "option"
                                            ]
                                    }
                                ],
                                "type": "basic"
                            }
                        ]
                    },
                    {
                        "description": "map option-tab to command-tab",
                        "manipulators": [
                            {
                                "from": {
                                    "key_code": "tab",
                                    "modifiers": {
                                        "mandatory": [
                                            "option"
                                        ],
                                        "optional": [
                                            "any"
                                        ]
                                    }
                                },
                                "to": [
                                    {
                                        "key_code": "tab",
                                        "modifiers": [
                                            "command"
                                            ]
                                    }
                                ],
                                "type": "basic"
                            }
                        ]
                    },
```

**Comment by slrgt** [comment]
@DDvO how can i do this for only a specific application? I want to swap ctrl and option but only in Blender

**Comment by tekezo** [comment]
You can use frontmost_application_if.
https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/conditions/frontmost-application/
